### PR TITLE
Add unittests for SpeedMeter

### PR DESCRIPTION
**Pull request by WTer**

### CODE REVIEW

File: p2p/speedmeter_test.go [hunk 1]
```diff
@@ -10,10 +10,48 @@ import (
...
-    sp.Feed(128 * 1024)
-    time.Sleep(300)
-    sp.Feed(128 * 1024)
-    fmt.Println("speedMeter:", sp.GetRate())
+func Test_NewSpeedMeter(t *testing.T) {
+    var step uint64 = 5
+    var itemsNum uint = 10
+    v1 := NewSpeedMeter(step, itemsNum)
+
+    if v1.step != step || v1.itemsNum != itemsNum {
```
Comment (lucusfly) [inline p2p/speedmeter_test.go]: use check.Assert?

File: p2p/speedmeter_test.go [hunk 2]
```diff
@@ -10,10 +10,48 @@ import (
...
+    }
+
+    rate := v1.GetRate()
+    if rate != 0 {
+        fmt.Println("NewSpeedMeter rate:", rate)
+        t.Fail()
+    }
+
+    var num uint = 10
+    time.Sleep(1000 * time.Millisecond)
```
Comment (lucusfly) [inline p2p/speedmeter_test.go]: why you need sleep?

File: p2p/speedmeter_test.go [hunk 3]
```diff
@@ -10,10 +10,48 @@ import (
...
+    if v1.preFeedTick <= 0 {
+        fmt.Println("NewSpeedMeter preFeedTick:", v1.preFeedTick)
+        t.Fail()
+    }
+
+    for _, ele := range v1.itemArr {
+        if ele.tick == 0 {
+            fmt.Println("NewSpeedMeter Feed:", ele)
+            t.Fail();
+        }
```
Comment (lucusfly) [inline p2p/speedmeter_test.go]: use check.Assert only need one line for 52-55 lines

File: p2p/speedmeter_test.go (hunk 2 above)
Comment (WTer) [reply p2p/speedmeter_test.go]: the following statement is Feed(), in which the time.Now will be called, so I use this simple way to simulate it.

File: p2p/speedmeter_test.go (hunk 1 above)
Comment (WTer) [reply p2p/speedmeter_test.go]: Currently I use testing.T.  we can refine it to check later.

**Pull request merged**